authors/apps/authentication/views.py

```diff
@@ -74,3 +84,83 @@ def update(self, request, *args, **kwargs):
 
         return Response(serializer.data, status=status.HTTP_200_OK)
 
+
+class ResetPasswordAPIView(GenericAPIView):
+    permission_classes = (AllowAny,)
+    serializer_class = ResetPasswordSerializer
+
+    @classmethod
+    def decode_id(self, uid):
+        username = urlsafe_base64_decode(uid).decode('utf-8')
+        return username
+        # print(username)
+
+    def post(self, request):
+        user = request.data.get('user', {})
+        serializer = self.serializer_class(data=user)
+        serializer.is_valid(raise_exception=True)
+        user = User.objects.filter(email=user['email']).first()
+        token = default_token_generator.make_token(user)
+        uid = urlsafe_base64_encode(force_bytes(user.username)).decode('utf-8')
+        domain = 'http://{}'.format(get_current_site(request))
+        route = 'api/password-reset-confirm/'
+        url = '{}/{}{}-{}'.format(domain, route, token, uid)
+        subject = 'Authors Haven: Have Your Password Changed'
+        message = 'Follow The Link Below To Reset Your Password {}'.format(url)
+        email_from = settings.EMAIL_HOST_USER
+        to_mail = user.email
+        recipient_list = [to_mail]
+        send_mail(
+            subject,
+            message,
+            email_from,
+            recipient_list,
+            fail_silently=False
+            )
+        res = {"message": "An Email Has Been Sent To This Email Address"}
+        return Response(res, status.HTTP_200_OK)
+
+
+class ResetPasswordConfirmAPIView(RetrieveUpdateAPIView):
+    permission_classes = (AllowAny,)
+    serializer_class = ResetPasswordConfirmSerializer
+
+    def get_object(self):
+        return {"password": "", "confirm_password": ""}
+
+    def update(self, request, **kwargs):
```

Comment:
@moseskamira Please reduce the size of this method , you could remove the first try and expect block and put it in a separate method called `validate_slug`.